cmd/finch/nerdctl_remote.go

```diff
@@ -501,3 +539,57 @@ func handleEnvFile(fs afero.Fs, systemDeps NerdctlCommandSystemDeps, arg, arg2 s
 	}
 	return skip, envs, nil
 }
+
+func handleDockerCompatInspect(fc *config.Finch, args []string) []string {
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" {
+		return args
+	}
+
+	newArgList := []string{}
+	isInspect := false
+	inspectIndex := -1
+	skipNext := false
+	inspectType := ""
+
+	for idx, arg := range args {
+		if skipNext {
+			skipNext = false
+			continue
+		}
+		if arg == "inspect" {
+			isInspect = true
+			inspectIndex = idx
+			newArgList = append(newArgList, arg)
+		}
+
+		if strings.Contains(arg, "--type") && (idx < len(args)-1) && isInspect {
```

Comment:
Should we be using the slices package to check if the args slice has a string `--type` in case the `inspect` arg comes after the `--type` arg?

Reply:
for now resolving it,  for semantics, if its too concerning will fix in my pr which will be based on top of it.